stubs/translationstring/translationstring/__init__.pyi

```diff
@@ -0,0 +1,81 @@
+from collections.abc import Callable
+from gettext import NullTranslations
+from typing import Any, Protocol
+from typing_extensions import Self
+
+class _TranslationStringFactory(Protocol):
+    def __call__(
+        self,
+        msgid: str | TranslationString,
+        mapping: dict[str, Any] | None = ...,
+        default: str | None = ...,
+        context: str | None = ...,
+    ) -> TranslationString: ...
+
+class _ChameleonTranslate(Protocol):
+    def __call__(
+        self,
+        msgid: str | TranslationString,
+        mapping: dict[str, Any] | None = ...,
+        context: str | None = ...,
+        target_language: str | None = ...,
+        default: str | None = ...,
+    ) -> TranslationString: ...
+
+class _TranslatorPolicy(Protocol):
+    def __call__(self, translations: NullTranslations, tstring: str, domain: str | None, context: str | None) -> str: ...
+
+class _Translator(Protocol):
+    def __call__(
+        self,
+        tstring: str | TranslationString,
+        domain: str | None = ...,
+        mapping: dict[str, Any] | None = ...,
+        context: str | None = ...,
+    ) -> str: ...
+
+class _PluralizerPolicy(Protocol):
+    def __call__(
+        self, translations: NullTranslations, singular: str, plural: str, n: int, domain: str | None, context: str | None
+    ) -> str: ...
+
+class _Pluralizer(Protocol):
+    def __call__(
+        self,
+        singular: str | TranslationString,
+        plural: str | TranslationString,
+        n: int,
+        domain: str | None = ...,
+        mapping: dict[str, Any] | None = ...,
+        context: str | None = ...,
+    ) -> str: ...
+
+class TranslationString(str):
+    domain: str | None
+    context: str | None
+    default: str
+    mapping: dict[str, Any] | None
+    def __new__(
+        self,
+        msgid: str | Self,
```

Comment:
As `Self` is a subclass of `str`:

```suggestion
        msgid: str,
```

Reply:
Absolutely correct, but since translationstring does special case instances of `TranslationString` in most function/method parameters that accept a string, I think it's a better to be explicit as a hint for the user, even if it is redundant as far as type checking is concerned.